verilog/formatting/tree_unwrapper.cc

```diff
@@ -1047,6 +1088,36 @@ static void IndentBetweenUVMBeginEndMacros(TokenPartitionTree* partition_ptr,
   }
 }
 
+// Try to merge object reference with method/function call, e.g.
+// ...
+//  { (>>>>[foo], policy: fit-else-expand) }
+//  { (>>>>[. bar ( foobar )], policy: append-fitting-sub-partitions)
+//    { (>>>>[. bar (], policy: append-fitting-sub-partitions) }
+// ...
+static void ReshapeMethodCall(const SyntaxTreeNode& node,
+                              TokenPartitionTree* partition_ptr) {
+  if (node.children().size() != 2) {
+    return ;
+  }
+
+  const SyntaxTreeNode& method_call_node =
+      SymbolCastToNode(*ABSL_DIE_IF_NULL(node.children().back()));
+  if (!method_call_node.MatchesTagAnyOf({NodeEnum::kMethodCallExtension})) {
+    return ;
+  }
+
+  auto& partition = *ABSL_DIE_IF_NULL(partition_ptr);
+  CHECK_GE(partition.Children().size(), 2);
+
+  auto& object_name = *ABSL_DIE_IF_NULL(partition.LeftmostDescendant());
```

Comment:
The invoking "object" could be more complex, and not necessarily a leaf partition.  (perhaps call this `object_tail`?)
`foo` could be a `foo[...]` or `foo(...)` or `(foo)`.

I'm concerned that `partition.LeftmostDescendant()` might not refer to the partition node you intend to move, in the case of more complicated invoking objects.  What I did in `kDataDeclaration` for module instances is flatten once before manipulating.

As far as partitioning shape in the more complex cases, I'm thinking (`bar(...)`, `.foo(...)`), and (`bar[...]`, `.foo(...)`).